src/components/DataAccessLayer.ts

```diff
@@ -1,16 +1,32 @@
-import { resources } from "@/resources";
+import { resources as staticResources } from "@/resources";
 import { inject } from "vue";
 import { ResourceInSearch } from "@/types";
-const getApisFromHost = inject("$getApi");
 
-export const findData = (inputValue: string): ResourceInSearch[] => {
-  console.log("findData...called with ..", inputValue, getApisFromHost); //Todo: Remove once API integration is done
+const fetchAllData = async (
+  tag: string,
+): Promise<ResourceInSearch[] | null> => {
+  try {
+    const getApisFromHost = inject("$getApi") as string;
+    const apiData = await fetch(
+      getApisFromHost + "?" + new URLSearchParams({ tag }),
+    );
+    return await apiData.json();
+  } catch (err) {
+    alert("Error fetching data from the server, only display test data.");
+    return null;
```

Comment:
Yeah I'm not sure about returning null here. Also, has it been decided to use alerts for error management? Were options like toasts explored?

Reply:
This is for ensure project still running when API connecting lost, provide warning, while still displaying something?